src/lib.rs

```diff
@@ -379,22 +385,42 @@ fn extract_crate_names(
     Ok(root_pkg.into_iter().chain(dep_pkgs).collect())
 }
 
-fn extract_package_name(cargo_toml: &DocumentMut) -> Option<&str> {
-    cargo_toml.get("package")?.get("name")?.as_str()
+fn extract_package_name<'c>(cargo_toml: &'c Manifest) -> Option<&'c str> {
+    cargo_toml.package().map(|p| p.name())
 }
 
-fn target_dep_tables(cargo_toml: &DocumentMut) -> impl Iterator<Item = &Table> {
-    cargo_toml.get("target").into_iter().filter_map(Item::as_table).flat_map(|t| {
-        t.iter().map(|(_, value)| value).filter_map(Item::as_table).flat_map(dep_tables)
+fn target_dep_tables<'c>(
+    cargo_toml: &'c Manifest,
+) -> impl Iterator<Item = (&'c str, &'c Dependency<'c>)> {
+    cargo_toml.targets().into_iter().flat_map(|t| {
+        t.iter()
+            .map(|(_, t)| t)
+            .flat_map(|t| combined_dep_tables(t.dependencies(), t.dev_dependencies()))
     })
 }
 
-fn dep_tables(table: &Table) -> impl Iterator<Item = &Table> {
-    table
-        .get("dependencies")
+fn dep_tables<'c>(cargo_toml: &'c Manifest) -> impl Iterator<Item = (&'c str, &'c Dependency<'c>)> {
+    combined_dep_tables(cargo_toml.dependencies(), cargo_toml.dev_dependencies())
+}
+
+fn combined_dep_tables<'c>(
+    deps: Option<&'c Dependencies<'c>>,
+    dev_deps: Option<&'c Dependencies<'c>>,
+) -> impl Iterator<Item = (&'c str, &'c Dependency<'c>)> {
+    let mut deps = deps
         .into_iter()
-        .chain(table.get("dev-dependencies"))
-        .filter_map(Item::as_table)
+        .flat_map(|deps| deps.iter())
+        .chain(dev_deps.into_iter().flat_map(|deps| deps.iter()))
+        .collect::<Vec<_>>();
+    // Ensure renames (i-e deps with `package` key) are listed the last.
+    deps.sort_by(|(_, a), (_, b)| match (a.package(), b.package()) {
```

Comment:
Why do we need this?

Reply:
Because the using code relies on this. I added this after a test case was failing because it was getting the name rather than the rename (or was it the other way around? :thinking:).